tests/amazon/aws/hooks/test_s3_hooks.py

```diff
@@ -35,3 +41,368 @@ async def test_aws_base_hook_async_get_client_async_with_aws_secrets(mock_get_co
+async def test_s3_key_hook_is_keys_unchanged_exception(mock_list_keys, mock_client):
+    """
+    Test is_key_unchanged gives AirflowException
+    :return:
+    """
+    mock_list_keys.return_value = []
+
+    s3_hook_async = S3HookAsync(client_type="S3", resource_type="S3")
+
+    with pytest.raises(AirflowException):
+        await s3_hook_async.is_keys_unchanged(
+            client=mock_client.return_value,
+            bucket_name="test_bucket",
+            prefix="test",
+            inactivity_period=1,
+            min_objects=1,
+            previous_objects=set("test"),
+            inactivity_seconds=0,
+            allow_delete=False,
+            last_activity_time=None,
+        )
+
+
+@mock.patch("astronomer.providers.amazon.aws.triggers.s3.S3HookAsync.get_client_async")
+@mock.patch("astronomer.providers.amazon.aws.triggers.s3.S3HookAsync._list_keys")
+@pytest.mark.asyncio
+async def test_s3_key_hook_is_keys_unchanged_true(mock_list_keys, mock_client):
+    """
+    Test is_key_unchanged gives AirflowException
+    :return:
+    """
+    mock_list_keys.return_value = ["test_file"]
+
+    s3_hook_async = S3HookAsync(client_type="S3", resource_type="S3")
+    response = await s3_hook_async.is_keys_unchanged(
+        client=mock_client.return_value,
+        bucket_name="test_bucket",
+        prefix="test",
+        inactivity_period=1,
+        min_objects=1,
+        previous_objects=set("t"),
+        inactivity_seconds=0,
+        allow_delete=True,
+        last_activity_time=datetime(2020, 8, 14, 17, 19, 34),
+    )
+
+    assert response is True
+
+
+@mock.patch("astronomer.providers.amazon.aws.hooks.s3.S3HookAsync.get_client_async")
+@pytest.mark.asyncio
+async def test_s3_prefix_sensor_hook_list_prefixes(mock_client):
+    """
+    Test list_prefixes for a valid response
+    """
+    test_resp_iter = [{"CommonPrefixes": [{"Prefix": "async-prefix1/"}, {"Prefix": "async-prefix2/"}]}]
+    mock_paginator = mock.Mock()
+    mock_paginate = mock.MagicMock()
+    mock_paginate.__aiter__.return_value = test_resp_iter
+    mock_paginator.paginate.return_value = mock_paginate
+
+    s3_hook_async = S3HookAsync(client_type="S3", resource_type="S3")
+    mock_client.get_paginator = mock.Mock(return_value=mock_paginator)
+
+    actual_output = await s3_hook_async.list_prefixes(mock_client, "test_bucket", "test")
+    expected_output = ['async-prefix1/', 'async-prefix2/']
+    assert expected_output == actual_output
+
+
+@mock.patch("astronomer.providers.amazon.aws.hooks.s3.S3HookAsync.get_client_async")
+@mock.patch("astronomer.providers.amazon.aws.hooks.s3.S3HookAsync.list_prefixes")
+@pytest.mark.asyncio
+async def test_s3_prefix_sensor_hook_check_for_prefix(mock_list_prefixes, mock_client):
+    """
+    Test is_key_unchanged gives AirflowException
+    :return:
+    """
+    mock_list_prefixes.return_value = ['async-prefix1/', 'async-prefix2/']
+
+    s3_hook_async = S3HookAsync(client_type="S3", resource_type="S3")
```

Comment:
These three cases can be parameterised for the tests.

Reply:
this is completed